etl/ems_person_match/match_ems_to_people.py

```diff
@@ -0,0 +1,314 @@
+from itertools import groupby
+import logging
+
+from rapidfuzz import fuzz
+
+from utils.exceptions import EMSPersonIdError
+
+from utils.field_maps import (
+    EMS_POS_IN_VEHICLE_TO_CRIS_OCC_POS_MAP,
+    CRIS_MODE_CAT_TO_EMS_TRAVEL_MODE_MAP,
+)
+from utils.graphql import (
+    make_hasura_request,
+    GET_UNMATCHED_EMS_PCRS,
+    GET_UNMATCHED_CRASH_PEOPLE,
+    GET_EMS_PCRS_BY_INCIDENT_NUMBER,
+    UPDATE_EMS_PCR,
+)
+from utils.logging import get_logger
+from utils.match_rules import MATCH_RULES
+
+AGE_GAP_TOLERANCE = 3
+TRANSPORT_DEST_MATCH_MIN_MIN_SCORE = 90
+
+
+def is_sex_match(pcr, person):
+    if not person["gndr"] or not pcr["pcr_patient_gender"]:
+        return False
+    return person["gndr"]["label"].lower() == pcr["pcr_patient_gender"].lower()
+
+
+def is_ethnicty_match(pcr, person):
+    if not person["drvr_ethncty"] or not pcr["pcr_patient_race"]:
+        return
+    return (
+        person["drvr_ethncty"]["label"].lower() in pcr["pcr_patient_race"].lower()
+    ) or (
+        "indian" in person["drvr_ethncty"]["label"].lower()
+        and "indian" in pcr["pcr_patient_race"].lower()
+    )
+
+
+def is_age_match(pcr, person):
+    if pcr["pcr_patient_age"] is None or person["prsn_age"] is None:
+        return False
+    return pcr["pcr_patient_age"] == person["prsn_age"]
+
+
+def is_approx_age_match(pcr, person):
+    if pcr["pcr_patient_age"] is None or person["prsn_age"] is None:
+        return False
+    return abs(pcr["pcr_patient_age"] - person["prsn_age"]) <= AGE_GAP_TOLERANCE
+
+
+def is_position_in_vehicle_match(pcr, person):
+    pcr_position = pcr["mvc_form_position_in_vehicle"]
+    person_position = person.get("occpnt_pos", {}).get("label", "")
+    if pcr_position and person_position:
+        pcr_position = pcr_position.lower()
+        person_position = person_position.lower()
+        pcr_position_mapped = EMS_POS_IN_VEHICLE_TO_CRIS_OCC_POS_MAP.get(pcr_position)
+        if pcr_position_mapped:
+            return pcr_position_mapped == person_position
+    return False
+
+
+def is_mode_match(pcr, person):
+    pcr_mode = pcr["travel_mode"]
+    person_mode = person["mode_desc"]
+    if pcr_mode and person_mode:
+        pcr_mode = pcr_mode.lower()
+        person_mode = person_mode.lower()
+        person_mode_mapped = CRIS_MODE_CAT_TO_EMS_TRAVEL_MODE_MAP.get(person_mode)
```

Comment:
If CRIS adds a new mode will this fail? Do we want this to fail?

Reply:
Also, if EMS changes their mode definitions will we notice that our matches have degraded?